MTA-6757 Formatting Fixes (#353)
migtools/mta-documentation: docs/topics/rules-development/yaml-dotnet-provider.adoc

:_mod-docs-content-type: REFERENCE
[id="yaml-dotnet-provider_{context}"]
= Csharp provider

[role="_abstract"]
The `csharp` provider is an external provider used to analyze `.NET Core` and `C#` source code. Currently, the provider supports the `referenced` capability.

The `csharp` provider uses a gRPC interface to perform a semantic analysis of an application source code in the `source-only` mode. The provider parses the source code by using tree-sitter and uses stack graph for the analysis to find references to types, methods, classes, and fields. Based on the custom rule definition, the analyzer identifies violations in your code that you must resolve before the application migration. 

== `referenced`

> The `csharp` provider supports `referenced` capability in rules to define fields such as `pattern` and `location` based on which the provider searches the code for violations. 

Reviewer: A missing article is there + we can make the sentence a bit easier in terms of readability, for example:
```suggestion
The `csharp` provider supports the `referenced` capability in rules to define fields, such as `pattern` and `location`, that the provider uses to search the code for violations.
```

Alternatively:
```suggestion
The `csharp` provider supports the `referenced` capability in rules to define fields such as `pattern` and `location`. The provider uses these fields to search the code for violations.
```

Author: Modified, thanks!
